packages/core/src/utils/mcp-server/utils.ts

```diff
@@ -0,0 +1,448 @@
+  if (transportName.includes('stdio')) {
+    return { mcpTransport: 'stdio', networkTransport: 'pipe' };
+  }
+  
+  // Streamable HTTP is the standard HTTP-based transport
+  // The official SDK uses 'StreamableHTTPServerTransport' / 'StreamableHTTPClientTransport'
+  if (transportName.includes('streamablehttp') || transportName.includes('streamable')) {
+    return { mcpTransport: 'http', networkTransport: 'tcp' };
+  }
+  
+  // SSE is the deprecated HTTP+SSE transport (backwards compatibility)
+  // Note: Modern Streamable HTTP can use SSE internally, but SSE transport is deprecated
+  if (transportName.includes('sse')) {
+    return { mcpTransport: 'sse', networkTransport: 'tcp' };
+  }
+
+  // For custom transports, mark as unknown
+  // TODO(bete): Add support for custom transports
+  return { mcpTransport: 'unknown', networkTransport: 'unknown' };
+}
+
+/** Extracts additional attributes for specific notification types */
+function getNotificationAttributes(
+  method: string,
+  params: Record<string, unknown>,
+): Record<string, string | number> {
+  const attributes: Record<string, string | number> = {};
+
+  switch (method) {
+    case 'notifications/cancelled':
+      if (params?.requestId) {
+        attributes['mcp.cancelled.request_id'] = String(params.requestId);
+      }
+      if (params?.reason) {
+        attributes['mcp.cancelled.reason'] = String(params.reason);
+      }
+      break;
+
+    case 'notifications/message':
+      if (params?.level) {
+        attributes['mcp.logging.level'] = String(params.level);
+      }
+      if (params?.logger) {
+        attributes['mcp.logging.logger'] = String(params.logger);
+      }
+      if (params?.data !== undefined) {
+        attributes['mcp.logging.data_type'] = typeof params.data;
+        // Store the actual message content
+        if (typeof params.data === 'string') {
+          attributes['mcp.logging.message'] = params.data;
+        } else {
+          attributes['mcp.logging.message'] = JSON.stringify(params.data);
+        }
+      }
+      break;
+
+    case 'notifications/progress':
+      if (params?.progressToken) {
+        attributes['mcp.progress.token'] = String(params.progressToken);
+      }
+      if (typeof params?.progress === 'number') {
+        attributes['mcp.progress.current'] = params.progress;
+      }
+      if (typeof params?.total === 'number') {
+        attributes['mcp.progress.total'] = params.total;
+        if (typeof params?.progress === 'number') {
+          attributes['mcp.progress.percentage'] = (params.progress / params.total) * 100;
+        }
+      }
+      if (params?.message) {
+        attributes['mcp.progress.message'] = String(params.message);
+      }
+      break;
+
+    case 'notifications/resources/updated':
+      if (params?.uri) {
+        attributes['mcp.resource.uri'] = String(params.uri);
+        // Extract protocol from URI
+        try {
+          const url = new URL(String(params.uri));
```

Comment:
we have a url parsing helper in the core package that we should use here. It also handles relative URLs.